a3_finance/a3_finance/doc_events/supplier.py

```
@@ -0,0 +1,14 @@
+
+def before_insert(doc, method):
+    today = getdate(nowdate())
+    if today < getdate(doc.custom_effective_from):
+        doc.disabled = 1
+
+def change_supplier_status():
+    today = getdate(nowdate())
+    supplier_list = frappe.db.get_all("Supplier",["name","custom_effective_from"])
+    for supplier in supplier_list:
+        if today >= getdate(supplier.custom_effective_from):
+            frappe.db.set_value("Supplier",supplier.name,"disabled",0)
```

Ruff (0.12.2)

No newline at end of file

Add trailing newline

(W292)